tests/basic/sharedv4_svc_test.go

```diff
@@ -1055,6 +1060,121 @@ var _ = Describe("{Sharedv4SvcFunctional}", func() {
+							Step(fmt.Sprintf("wait for app %s to have 0 pods", ctx.App.Key), func() {
+								waitForNumPodsToEqual(ctx, 0)
+							})
+
+							Step(fmt.Sprintf("scale it backsssss %s", ctx.App.Key), func() {
+								scaleApp(ctx, numPods)
+
+								ValidateContext(ctx)
+							})
+
+							// validate the pods are all on one node
+							Step(fmt.Sprintf("validate that all pods are running on the attached node for %s", ctx.App.Key), func() {
+								pods, err := core.Instance().GetPodsUsingPV(vol.ID)
+								Expect(err).NotTo(HaveOccurred())
+								for _, pod := range pods {
+									Expect(pod.Spec.NodeName).To(Equal(attachedNode.Name))
+								}
+
+							})
+						})
+				}
+
+			})
+
+			It("should set device path to RO and validate recovery", func() {
+				for _, ctx := range testSharedV4Contexts {
+					_, apiVol, attachedNode := getSv4TestAppVol(ctx)
+					counterCollectionInterval := 3 * time.Duration(numPods) * time.Second
+					devicePath := fmt.Sprintf("%s%s", devicePathPrefix, apiVol.Id)
+
+					// In porx, we check the export path is read-only when it is previously RO and
+					// changed to RW by fs due to occuring error. Specifically, when mnt.Opts (default)
+					// is RW and mnt.VfsOpts (fs state) is RO.
+					// By setting device path to RO, we mock the same behavior. Check state in
+					// `/proc/self/mountinfo`
+					Step(fmt.Sprintf("mark device path as RO %s", ctx.App.Key), func() {
+						setPathToROMode(devicePath, attachedNode)
+					})
+
+					Step(fmt.Sprintf("validate the counters are inactive %s", ctx.App.Key), func() {
+						counters := getAppCounters(apiVol, attachedNode, counterCollectionInterval)
+						activePods := getActivePods(counters)
+						Expect(len(activePods)).To(Equal(0))
+					})
+
+					Step(fmt.Sprintf("restart vol driver %s", ctx.App.Key), func() {
+						restartVolumeDriverOnNode(attachedNode)
+					})
+
+					Step(fmt.Sprintf("validate counter are active for %s", ctx.App.Key), func() {
+						counters := getAppCounters(apiVol, attachedNode, counterCollectionInterval)
+						activePods := getActivePods(counters)
+						Expect(len(activePods)).To(Equal(numPods))
+					})
+
+					Step(fmt.Sprintf("validate device path is set as RW for %s", ctx.App.Key), func() {
+						mntList, err := mount.GetMounts()
+						Expect(err).NotTo(HaveOccurred())
+						var volumeMountRW = regexp.MustCompile(`,rw,|,rw|rw,|rw`)
+
+						for _, mnt := range mntList {
+							if mnt.Mountpoint != devicePath {
+								continue
+							}
+							Expect(volumeMountRW.MatchString(mnt.Opts)).To(BeTrue())
+							Expect(volumeMountRW.MatchString(mnt.VfsOpts)).To(BeTrue())
+						}
+					})
+				}
+			})
+
+			JustAfterEach(func() {
+				for _, ctx := range testSharedV4Contexts {
+					_, _, attachedNode := getSv4TestAppVol(ctx)
+					Step(fmt.Sprintf("remove label on node for %s", ctx.App.Key), func() {
+						err := core.Instance().RemoveLabelOnNode(attachedNode.Name, "attachedNode")
```

Comment:
since "attachedNode" is being used in more than one place, it should be a const

Reply:
do you mean to set it right at the beginning of context? if so, isn't it going to change for each additional context we run with?

Reply:
"attachedNode" is removed since we don't need to run on single node.